backend.go

```diff
@@ -255,6 +365,76 @@ func (b *kubeAuthBackend) role(ctx context.Context, s logical.Storage, name stri
 	return role, nil
 }
 
+// getHTTPClient return the backend's HTTP client for connecting to the Kubernetes API.
+func (b *kubeAuthBackend) getHTTPClient() (*http.Client, error) {
+	if b.httpClient == nil {
+		return nil, fmt.Errorf("the backend's http.Client has not been initialized")
+	}
+
+	if b.tlsConfig == nil {
+		return nil, fmt.Errorf("the backend's TLSConfig has not been initialized")
+	}
+
+	return b.httpClient, nil
+}
+
+// updateTLSConfig ensures that the httpClient's TLS configuration is consistent
+// with the backend's stored configuration.
+func (b *kubeAuthBackend) updateTLSConfig(config *kubeConfig) error {
+	b.tlsMu.Lock()
+	defer b.tlsMu.Unlock()
+
+	if b.httpClient == nil {
+		return fmt.Errorf("the backend's http.Client has not been initialized")
+	}
+
+	// attempt to read the CA certificates from the config directly or from the filesystem.
+	var caCertBytes []byte
+	if config.CACert != "" {
+		caCertBytes = []byte(config.CACert)
+	} else if !config.DisableLocalCAJwt && b.localCACertReader != nil {
+		// TODO: this may block on I/O, investigate a proper mitigation
+		data, err := b.localCACertReader.ReadFile()
+		if err != nil {
+			return err
+		}
+		caCertBytes = []byte(data)
+	}
+
+	transport, ok := b.httpClient.Transport.(*http.Transport)
+	if !ok {
+		// should never happen
+		return fmt.Errorf("type assertion failed for %T", b.httpClient.Transport)
+	}
```

Comment:
nit: This variable is only used inside the `if !b.tlsConfig.RootCAs.Equal(certPool) {` block below, so it can be scoped tighter by moving it down there.

Reply:
No qualms here. Originally this check was meant to prevent needlessly processing of the certificates.

Addressed in 89ae775c1a646e242358166315c32efd4c86656a